keras_cv/models/feature_extractor/clip/clip_model.py

```diff
@@ -0,0 +1,189 @@
+            transformer-based text encoder.
+        transformer_layers (int): The number of layers in the transformer-based
+            text encoder.
+    """
+
+    def __init__(
+        self,
+        embed_dim=512,
+        image_resolution=224,
+        vision_layers=12,
+        vision_width=768,
+        vision_patch_size=32,
+        context_length=77,
+        vocab_size=49408,
+        transformer_width=768,
+        transformer_heads=8,
+        transformer_layers=12,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        try:
+            import keras_nlp  # noqa: F401
+        except ImportError:
+            raise ImportError(
+                "CLIP model requires keras-nlp. Please pip "
+                "install keras-nlp."
+            )
+        self.embed_dim = embed_dim
+        self.image_resolution = image_resolution
+        self.vision_layers = vision_layers
+        self.vision_width = vision_width
+        self.vision_patch_size = vision_patch_size
+        self.context_length = context_length
+        self.vocab_size = vocab_size
+        self.transformer_width = transformer_width
+        self.transformer_heads = transformer_heads
+        self.transformer_layers = transformer_layers
+
+        vision_heads = self.vision_width // 64
+        self.image_encoder = CLIPImageEncoder(
+            input_resolution=self.image_resolution,
+            patch_size=self.vision_patch_size,
+            width=self.vision_width,
+            num_layers=self.vision_layers,
+            heads=vision_heads,
+            output_dim=self.embed_dim,
+            name="image_encoder",
+        )
+        self.text_encoder = CLIPTextEncoder(
+            transformer_width=self.transformer_width,
+            transformer_layers=self.transformer_layers,
+            transformer_heads=self.transformer_heads,
+            vocab_size=self.vocab_size,
+            embed_dim=self.embed_dim,
+            context_length=self.context_length,
+            name="text_encoder",
+        )
+
+        self.logit_scale = keras.Variable(
+            ops.ones([]) * ops.log(1 / 0.07), name="logit_scale"
+        )
+        self.image_embeddings = None
+        self.text_embeddings = None
+
+    def build(self, input_shape):
+        super().build(input_shape)
+        self.text_encoder.build([None, self.context_length])
+        self.image_encoder.build(
+            [None, self.image_resolution, self.image_resolution, 3]
+        )
+
+    def encode_images(self, image):
+        return self.image_encoder(image)
+
+    def encode_text(self, text, attention_mask=None):
+        return self.text_encoder(text, attention_mask=attention_mask)
+
+    def call(self, image, text, attention_mask=None):
+        self.image_embeddings = self.encode_images(image)
+        self.text_embeddings = self.encode_text(
+            text, attention_mask=attention_mask
+        )
+        normalize_image_features = keras.ops.sqrt(
+            keras.ops.sum(
```

Comment:
We have imported `ops`, can we directly use `ops.sum` and similarly for other places in this file where we use `keras.ops.`?